extensions/kafka-client/runtime/src/main/java/io/quarkus/kafka/client/runtime/graal/Crc32CSubstitutions.java

```diff
@@ -0,0 +1,35 @@
+package io.quarkus.kafka.client.runtime.graal;
+
+import java.lang.invoke.MethodHandle;
+import java.util.zip.Checksum;
+
+import org.apache.kafka.common.utils.Crc32C;
+
+import com.oracle.svm.core.annotate.Alias;
+import com.oracle.svm.core.annotate.RecomputeFieldValue;
+import com.oracle.svm.core.annotate.RecomputeFieldValue.Kind;
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+import com.oracle.svm.core.jdk.JDK11OrLater;
+
+/**
+ * The following substitution replaces the usage of {@code MethodHandle} in {@code Java9ChecksumFactory} with a plain
+ * constructor invocation when run under GraalVM. This is necessary because the native image generator does not support method
+ * handles.
+ */
+@TargetClass(value = Crc32C.class, innerClass = "Java9ChecksumFactory", onlyWith = JDK11OrLater.class)
```

Comment:
Hello @gwenneg, since the `Java9ChecksumFactory` gets used for any version of Java 9 or more[1], this substitution's `onlyWith` should apply for Java version 9 and greater, isnt' it?

[1] https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/utils/Crc32C.java#L40

Reply:
Thanks for reviewing this @jaikiran. I asked myself the same question, but GraalVM only supports JDK 8 and 11 AFAIK so it doesn't really make sense to use an `onlyWith` condition checking for Java 9 and greater. WDYT?

Reply:
@gwenneg I hadn't thought of that aspect :) So yes, you are right, what you have here is fine IMO.